File: cpp/05_020.cpp
#include <queue>
#include <string>

class TaskScheduler {
private:
    std::queue<std::string> task_queue;
        
public:
    void add_task(const std::string& task) {
        task_queue.push(task);
    }
        
    void process_tasks() {
        while (!task_queue.empty()) {
            std::string current_task = task_queue.front();
            task_queue.pop();
            // Process task here
        }
    }
};

File: java/05_020.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class TaskScheduler {
        private Deque<String> taskQueue;
        
        public TaskScheduler() {
            this.taskQueue = new ArrayDeque<>();
        }
        
        public void addTask(String task) {
            this.taskQueue.add(task);
        }
        
        public void processTasks() {
            while (!this.taskQueue.isEmpty()) {
                String currentTask = this.taskQueue.pollFirst();
                // Process task here
            }
        }
    }
}